Add Foreword

zygoloid opened the pull request:
The general text was provided by the ISO CS. A list of incorporated
technical specifications was derived from N4824. A list of technical
changes since C++17 was derived from P2131R0.

Fixes #3668.

tkoeppe commented on source/preface.tex:
```diff
@@ -0,0 +1,184 @@
+\item ISO/IEC TS 21425:2017 \textit{Programming Languages --- \Cpp{} Extensions for ranges},
+\item ISO/IEC TS 22277:2017 \textit{Programming Languages --- \Cpp{} Extensions for Coroutines},
+\item ISO/IEC TS 21544:2018 \textit{Programming Languages --- Extensions to \Cpp{} for Modules},
+\item portions of ISO/IEC TS 19571:2016 \textit{Programming Languages --- Technical specification for \Cpp{} extensions for concurrency}, and % latch and barrier
```
Why does this entry get an explanatory comment, and the next one does not?

zygoloid replied:
Only because I found it non-obvious whether we took anything from this one, and easier to determine we took things from Lib Fundamentals.

tkoeppe replied:
OK :-)

tkoeppe commented on source/preface.tex:
```diff
@@ -0,0 +1,184 @@
+\item addition of \keyword{using} \keyword{enum} declaration
+\item addition of \keyword{char8_t} type
+\item guarantee that \keyword{char16_t} and \keyword{char32_t} literals
+are encoded as UTF-16 and UTF-32 respectively
```
Should there be a comma before "respectively"?

zygoloid replied:
Added.

tkoeppe commented on source/preface.tex:
```diff
@@ -0,0 +1,184 @@
+\item addition of \keyword{char8_t} type
+\item guarantee that \keyword{char16_t} and \keyword{char32_t} literals
+are encoded as UTF-16 and UTF-32 respectively
+\item support for an initializer statement in range-based for loops
```
Missing code font formatting for for formulation

zygoloid replied:
OK. Also loops -> statements, per [stmt.ranged].

tkoeppe commented on source/preface.tex:
```diff
@@ -0,0 +1,184 @@
+    \tcode{type_identity},
+    \tcode{unwrap_ref_decay}, and
+    \tcode{unwrap_reference}
+  \item new standard library primitive functions \tcode{assume_aligned} and
```
(Is "primitive" a good word for this?)

zygoloid replied:
Ehh. I was thinking "things that can't be implemented in the language", but perhaps that's just not an interesting distinction to be drawing here. Simplified.

zygoloid commented:
> I thought we already had this earlier?

Some preliminary review was done (but those were comments left on a random commit, which github is not well-suited to preserving). Now I'm asking for signoff on the result of addressing those comments.
